### PR TITLE
Update build/install-gentoo instructions with updates from build/install

### PR DESCRIPTION
**Pull request by smoke**
This merge request updates build/install-gentoo with what was changed in build/install so that Phalcon compiles on Gentoo again.

### CODE REVIEW

File: build/install-gentoo [hunk 1]
```diff
@@ -16,27 +16,54 @@
...
+
+PHP_FULL_VERSION=`php-config --version`
+if [ $? != 0 ]; then
+	echo "php-config is not installed"
+	exit 1
+fi
+
+if [ "${PHP_FULL_VERSION:0:3}" == "5.3" ]; then
+	echo "php 5.3 is no longer supported"
+	exit 1
```
Comment (Jurigag) [inline build/install-gentoo]: Not sure if we want to exit, phalcon as far as i know can still compile with no problem on php 5.3 and 5.4 ? Or no ?

Comment (smoke) [inline build/install-gentoo]: @Jurigag This is what the original code in build/install for building 3.0.x version does, so I have just ported it over. I have not tested compiling it or anything on PHP 5.3 and 5.4

Comment (Jurigag) [inline build/install-gentoo]: Oh okay, didn't know about it.

**Comment by andresgutierrez** [comment]
I actually don't like the solution, you can compile a sample file and test whether the flags are ok and then fall back if required.

**Comment by sergeyklay** [comment]
Or just use http://www.pixelbeat.org/scripts/gcccpuopt

**Comment by smoke** [comment]
I see, but just wondering why aren't these flags detections part of the original build/install but only build/install-gentoo?

**Comment by andresgutierrez** [comment]
What error are you getting when running the standard `install` script?

**Comment by smoke** [comment]
Here is the full output of the `build/install` script
```
find . -name \*.gcno -o -name \*.gcda | xargs rm -f
find . -name \*.lo -o -name \*.o | xargs rm -f
find . -name \*.la -o -name \*.a | xargs rm -f 
find . -name \*.so | xargs rm -f
find . -name .libs -a -type d|xargs rm -rf
rm -f libphp.la       modules/* libs/*
Cleaning..
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for icc... no
checking for suncc... no
checking whether gcc and cc understand -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for PHP prefix... /usr/lib64/php5.6
checking for PHP includes... -I/usr/lib64/php5.6/include/php -I/usr/lib64/php5.6/include/php/main -I/usr/lib64/php5.6/include/php/TSRM -I/usr/lib64/php5.6/include/php/Zend -I/usr/lib64/php5.6/include/php/ext -I/usr/lib64/php5.6/include/php/ext/date/lib
checking for PHP extension directory... /usr/lib64/php5.6/lib/extensions/no-debug-non-zts-20131226
checking for PHP installed headers prefix... /usr/lib64/php5.6/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.13.7.5 (ok)
checking for gawk... gawk
checking whether to enable phalcon... yes, shared
checking whether HAVE_BUNDLED_PCRE is declared... no
checking whether HAVE_JSON is declared... yes
checking for ext/json/php_json.h... yes
checking for ld used by gcc... /usr/x86_64-pc-linux-gnu/bin/ld
checking if the linker (/usr/x86_64-pc-linux-gnu/bin/ld) is GNU ld... yes
checking for /usr/x86_64-pc-linux-gnu/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 1572864
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC
checking if gcc PIC flag -fPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking whether the gcc linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: creating ./config.status
config.status: creating config.h
/bin/sh /var/www/smoke/phalcon/cphalcon/build/php5/64bits/libtool --mode=compile gcc  -I. -I/var/www/smoke/phalcon/cphalcon/build/php5/64bits -DPHP_ATOM_INC -I/var/www/smoke/phalcon/cphalcon/build/php5/64bits/include -I/var/www/smoke/phalcon/cphalcon/build/php5/64bits/main -I/var/www/smoke/phalcon/cphalcon/build/php5/64bits -I/usr/lib64/php5.6/include/php -I/usr/lib64/php5.6/include/php/main -I/usr/lib64/php5.6/include/php/TSRM -I/usr/lib64/php5.6/include/php/Zend -I/usr/lib64/php5.6/include/php/ext -I/usr/lib64/php5.6/include/php/ext/date/lib  -DPHALCON_RELEASE -DHAVE_CONFIG_H  -march=native -mtune=native -O2 -fomit-frame-pointer -fvisibility=hidden   -c /var/www/smoke/phalcon/cphalcon/build/php5/64bits/phalcon.zep.c -o phalcon.lo 
libtool: Version mismatch error.  This is libtool 2.4.6, but the
libtool: definition of this LT_INIT comes from an older release.
libtool: You should recreate aclocal.m4 with macros from libtool 2.4.6
libtool: and run autoconf again.
Makefile:181: recipe for target 'phalcon.lo' failed
make: *** [phalcon.lo] Error 63
```

**Comment by andresgutierrez** [comment]
It's asking for run autoconf after phpize, can you try:

```bash
cd cphalcon/ext
phpize
autoreconf --force --install
./configure --enable-phalcon
make
sudo make install
```

**Comment by smoke** [comment]
@andresgutierrez No success:
```
smoke@viper-pc /var/www/smoke/cphalcon/ext $ phpize 
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226
smoke@viper-pc /var/www/smoke/cphalcon/ext $ autoreconf --force --install
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: You should add the contents of the following files to 'aclocal.m4':
libtoolize:   '/usr/share/aclocal/libtool.m4'
libtoolize:   '/usr/share/aclocal/ltoptions.m4'
libtoolize:   '/usr/share/aclocal/ltsugar.m4'
libtoolize:   '/usr/share/aclocal/ltversion.m4'
libtoolize:   '/usr/share/aclocal/lt~obsolete.m4'
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.in,
libtoolize: and rerunning libtoolize and aclocal.
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
smoke@viper-pc /var/www/smoke/cphalcon/ext $ ./configure --enable-phalcon
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for PHP prefix... /usr/lib64/php5.6
checking for PHP includes... -I/usr/lib64/php5.6/include/php -I/usr/lib64/php5.6/include/php/main -I/usr/lib64/php5.6/include/php/TSRM -I/usr/lib64/php5.6/include/php/Zend -I/usr/lib64/php5.6/include/php/ext -I/usr/lib64/php5.6/include/php/ext/date/lib
checking for PHP extension directory... /usr/lib64/php5.6/lib/extensions/no-debug-non-zts-20131226
checking for PHP installed headers prefix... /usr/lib64/php5.6/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.13.7.5 (ok)
checking for gawk... gawk
checking whether to enable phalcon... yes, shared
checking whether HAVE_BUNDLED_PCRE is declared... no
checking whether HAVE_JSON is declared... yes
checking for ext/json/php_json.h... yes
checking for ld used by cc... /usr/x86_64-pc-linux-gnu/bin/ld
checking if the linker (/usr/x86_64-pc-linux-gnu/bin/ld) is GNU ld... yes
checking for /usr/x86_64-pc-linux-gnu/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 1572864
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC
checking if cc PIC flag -fPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: creating ./config.status
config.status: creating config.h
smoke@viper-pc /var/www/smoke/cphalcon/ext $ make
/bin/sh /var/www/smoke/cphalcon/ext/libtool --mode=compile cc  -I. -I/var/www/smoke/cphalcon/ext -DPHP_ATOM_INC -I/var/www/smoke/cphalcon/ext/include -I/var/www/smoke/cphalcon/ext/main -I/var/www/smoke/cphalcon/ext -I/usr/lib64/php5.6/include/php -I/usr/lib64/php5.6/include/php/main -I/usr/lib64/php5.6/include/php/TSRM -I/usr/lib64/php5.6/include/php/Zend -I/usr/lib64/php5.6/include/php/ext -I/usr/lib64/php5.6/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/www/smoke/cphalcon/ext/phalcon.c -o phalcon.lo 
libtool: Version mismatch error.  This is libtool 2.4.6, but the
libtool: definition of this LT_INIT comes from an older release.
libtool: You should recreate aclocal.m4 with macros from libtool 2.4.6
libtool: and run autoconf again.
Makefile:181: recipe for target 'phalcon.lo' failed
make: *** [phalcon.lo] Error 63
```

**Comment by sergeyklay** [comment]
@smoke I've fixed `build/install` script to work with Gentoo and macOS. 
Please check the `3.1.x` branch and re-open this if you continue experiencing problems compiling.

**Comment by smoke** [comment]
@sergeyklay Thanks!

**Pull request closed without merging**